devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1366,116 +1522,154 @@ void YarpRobotLoggerDevice::run()
                 {
-                    signalFullName = signal.signalName + "::" + key;
+                    signalName = signal.signalName + TREE_DELIM + key;
                     const auto& metadata = signal.metadata.vectors.find(key);
                     if (metadata == signal.metadata.vectors.cend())
                     {
                         log()->warn("{} Unable to find the metadata for the signal named {}. The "
                                     "default one will be used.",
                                     logPrefix,
-                                    signalFullName);
-                        m_bufferManager.addChannel({signalFullName, {vector.size(), 1}});
+                                    signalName);
+                        (*this.*initMetadataFunction)(signalName, {});
                     } else
                     {
-                        // if the metadata is found we use it
-                        m_bufferManager.addChannel({signalFullName, //
-                                                    {vector.size(), 1},
-                                                    metadata->second});
+                        (*this.*initMetadataFunction)(signalName, {metadata->second});
                     }
-
-                    signal.dataArrived = true;
                 }
+                signal.dataArrived = true;
+                signal.numMissedPackets = 0;
+                newMetadata = true;
             }
+            else
+            {
+                signal.numMissedPackets = 0;
+                for (const auto& [key, vector] : externalSignalCollection->vectors)
+                {
+                    signalName = signal.signalName + TREE_DELIM + key;
+                    m_bufferManager.push_back(vector, time, signalName);
+                    if(sendDataRT)
+                    {
+                        rtSignalName
+                            = ROBOT_RT_ROOT_NAME + TREE_DELIM + signalName;
+                        m_vectorCollectionRTDataServer.populateData(rtSignalName, vector);
+                    }
+                }
 
-            for (const auto& [key, vector] : collection->vectors)
+            }
+        }
+        else if (externalSignalCollection == nullptr && signal.connected)
+        {
+            signal.numMissedPackets++;
+            if (signal.numMissedPackets == maxTimoutForExogenousSignal)
             {
-                signalFullName = signal.signalName + "::" + key;
-                m_bufferManager.push_back(vector, time, signalFullName);
+                signal.connected = false;
+                signal.numMissedPackets = 0;
             }
         }
     }
+    Eigen::Matrix<double, 1, 1> newMetadataVector;
+    if(newMetadata)
+        m_updatedMetadataSignalVal = (m_updatedMetadataSignalVal + 1) % 2;
+    newMetadataVector << m_updatedMetadataSignalVal;
+
+    m_vectorCollectionRTDataServer.populateData(ROBOT_RT_ROOT_NAME + TREE_DELIM + "newMetadata", newMetadataVector);
```

Comment:
It's not clear to me the what is the purpose of this code

Reply:
Better to discuss face-to-face. This code informs the robot-log-visualizer when new metadata has been added to the vectors collection thus allowing the visualizer to append the data to the variable tree so the user can visualize it. It is a flag